grammar.js

```diff
@@ -1094,6 +1115,94 @@ module.exports = grammar({
       ),
     )),
 
+    _with_settings: $ => seq(
+          field('name', $.identifier),
+          optional('='),
+          field('value', choice($.identifier, alias($._single_quote_string, $.literal))),
+    ),
+
+    create_database: $ => seq(
+      $.keyword_create,
+      $.keyword_database,
+      $.identifier,
+      optional($.keyword_with),
+      repeat(
+        $._with_settings
+      ),
+    ),
+
+    create_role: $ => seq(
+      $.keyword_create,
+      choice(
+        $.keyword_user,
+        $.keyword_role,
+        $.keyword_group,
+      ),
+      $.identifier,
+      optional($.keyword_with),
+      repeat(
+        choice(
+          $._user_access_role_config,
+          $._role_options,
+        ),
+      ),
+    ),
+
+    _role_options: $ => choice(
+      field("option", $.identifier),
+      seq(
+        $.keyword_valid,
+        $.keyword_until,
+        field("valid_until", alias($._literal_string, $.literal))
+      ),
+      seq(
+        $.keyword_connection,
+        $.keyword_limit,
+        field("connection_limit", alias($._integer, $.literal))
+      ),
+      seq(
+        optional($.keyword_encrypted),
+        $.keyword_password,
+        choice(
+          field("password", alias($._literal_string, $.literal)),
+          $.keyword_null,
+        ),
+      ),
+    ),
+
+    _user_access_role_config: $ => prec.left(seq(
```

Comment:
did you break this out of _role_options for the precedence? you should be able to apply `prec.left` to an individual element in a `choice` -- I did it the other day testing the `create schema` stuff

Reply:
I broke it out, because I was unhappy with more spaghetti code :-) I just realized that the prec.left is not necessary. I will remove it.